packages/target-nodejs-sdk/test/telemetry.spec.js

```diff
@@ -0,0 +1,422 @@
+  client: "someClientId",
+  organizationId: "someOrgId",
+  artifactFormat: "json", // setting this tells the artifactProvider deobfuscation is not needed
+  targetLocationHint: "28",
+  pollingInterval: 0,
+  maximumWaitReady: 500,
+  telemetryEnabled: true
+};
+
+describe("telemetry mode", () => {
+  let client;
+
+  beforeEach(() => {
+    fetch.resetMocks();
+    if (client) {
+      client = undefined;
+    }
+  });
+
+  describe("on-device", () => {
+    it("immediately executes telemetries", () => {
+      expect.assertions(1);
+
+      return new Promise(done => {
+        fetch
+          .once(JSON.stringify(DECISIONING_PAYLOAD_FEATURE_FLAG))
+          .once(JSON.stringify(DELIVERY_RESPONSE));
+
+        async function clientReady() {
+          await client.getOffers({
+            request: {
+              ...targetRequest,
+              prefetch: {
+                mboxes: [
+                  {
+                    name: "mbox-feature-flags",
+                    index: 1
+                  },
+                  {
+                    name: "remote-only-mbox-a",
+                    index: 2
+                  },
+                  {
+                    name: "remote-only-mbox-b",
+                    index: 2
+                  }
+                ]
+              }
+            },
+            sessionId: "dummy_session"
+          });
+
+          setTimeout(() => {
+            expect(executeTelemetries.mock.calls.length).toBe(1);
+            done();
+          }, 500);
+        }
+
+        client = TargetClient.create({
+          decisioningMethod: DECISIONING_METHOD.ON_DEVICE,
+          ...targetClientOptions,
+          events: { clientReady }
+        });
+      });
+    });
+  });
+
+  describe("hybrid", () => {
+    it("calls sendNotifications when on-device used", () => {
+      expect.assertions(2);
+
+      return new Promise(done => {
+        fetch
+          .once(JSON.stringify(DECISIONING_PAYLOAD_FEATURE_FLAG))
+          .once(JSON.stringify(DELIVERY_RESPONSE));
+
+        async function clientReady() {
+          const result = await client.getOffers({
+            request: {
+              ...targetRequest,
+              prefetch: {
+                mboxes: [
+                  {
+                    name: "mbox-feature-flags",
+                    index: 1
+                  }
+                ]
+              }
+            },
+            sessionId: "dummy_session"
+          });
+
+          expect(result.meta.decisioningMethod).toBe(
+            DECISIONING_METHOD.ON_DEVICE
+          );
+          setTimeout(() => {
+            expect(executeTelemetries.mock.calls.length).toBe(1);
+            done();
+          }, 500);
+        }
+
+        client = TargetClient.create({
+          decisioningMethod: DECISIONING_METHOD.HYBRID,
+          ...targetClientOptions,
+          events: { clientReady }
+        });
+      });
+    });
+
+    it("executes telemetries when server-side used", () => {
```

Comment:
Let's rename this to `executes telemetries on next getOffers call when server-side used` so it's a bit more clear